client/components/tinymce/plugins/mentions/suggestion.jsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+const Suggestion = React.createClass( {
+	displayName: 'Suggestion',
+
+	propTypes: {
+		avatarUrl: React.PropTypes.string,
+		username: React.PropTypes.string,
+		fullName: React.PropTypes.string,
+		query: React.PropTypes.string
+	},
+
+	getDefaultProps() {
+		return {
+			avatarUrl: '',
+			username: '',
+			fullName: '',
+			query: ''
+		};
+	},
+
+	getRegExpFor( type, textToHighlight ) {
+		const expressions = {};
+
+		expressions.username = '(^' + textToHighlight + ')(\\w*)\\s*';
+		expressions.fullName = '(^.*?)(\\b' + textToHighlight + ')(.*)';
+
+		return new RegExp( expressions[ type ], 'ig' );
+	},
+
+	highlight( content, textToHighlight, type ) {
```

Comment:
Since `highlight` has access to the `this.props.query` prop and the function is not called with any other value, do we need the second argument? I guess it makes it more generic, in which case maybe this would be more useful as a standalone utility library.